contentpal-testing/src/test/java/org/dmfs/android/contentpal/testing/predicates/PredicateMatcherTest.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.android.contentpal.testing.predicates;
+
+import org.dmfs.android.contentpal.Predicate;
+import org.dmfs.android.contentpal.TransactionContext;
+import org.dmfs.iterables.ArrayIterable;
+import org.dmfs.iterables.EmptyIterable;
+import org.dmfs.jems.hamcrest.matchers.AbsentMatcher;
+import org.dmfs.optional.Absent;
+import org.dmfs.optional.Present;
+import org.junit.Test;
+
+import static org.dmfs.android.contentpal.testing.predicates.PredicateMatcher.argumentValues;
+import static org.dmfs.android.contentpal.testing.predicates.PredicateMatcher.backReferences;
+import static org.dmfs.android.contentpal.testing.predicates.PredicateMatcher.emptyArguments;
+import static org.dmfs.android.contentpal.testing.predicates.PredicateMatcher.predicateWith;
+import static org.dmfs.android.contentpal.testing.predicates.PredicateMatcher.selection;
+import static org.dmfs.jems.hamcrest.matchers.PresentMatcher.isPresent;
+import static org.dmfs.jems.mockito.doubles.TestDoubles.dummy;
+import static org.dmfs.jems.mockito.doubles.TestDoubles.failingMock;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+
+/**
+ * Unit test for {@link PredicateMatcher}.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class PredicateMatcherTest
+{
+
+    @Test
+    public void testSelection()
+    {
+        // Without TC
+        selection("a").matches(
+                when(mock(Predicate.class).selection(any(TransactionContext.class))).thenReturn("a").getMock());
+
+        // With TC
+        TransactionContext dummyTc = dummy(TransactionContext.class);
+
+        assertTrue(selection(dummyTc, "a").matches(
```

Comment:
also do the negative check, i.e. is must return `false` if the arguments don't match.

Reply:
It occured to me, too, afterwards... added some assertions for `false` case as well.